custom/icds_reports/management/commands/fill_bihar_api_data.py

```diff
@@ -0,0 +1,96 @@
+        else:
+            icds_state_aggregation_task(state_id=STATE_ID, date=month,
+                                        func_name='_aggregate_bp_forms')
+
+    def build_pnc_data(self, month):
+        if month == '2020-03-01':
+            path = os.path.join(os.path.dirname(__file__), 'sql_scripts', 'build_pnc_initial_data.sql')
+            with open(path, "r", encoding='utf-8') as sql_file:
+                sql_to_execute = sql_file.read()
+                sql_to_execute = sql_to_execute.format(month=month.strftime("%Y-%m-%d"), state_id=STATE_ID)
+                _run_custom_sql_script(sql_to_execute)
+        else:
+            icds_state_aggregation_task(state_id=STATE_ID, date=month,
+                                        func_name='_aggregate_ccs_record_pnc_forms')
+
+    def build_migration_forms(self, month):
+        if month == '2020-03-01':
+            path = os.path.join(os.path.dirname(__file__), 'sql_scripts', 'build_migration_initial_data.sql')
+            with open(path, "r", encoding='utf-8') as sql_file:
+                sql_to_execute = sql_file.read()
+                sql_to_execute = sql_to_execute.format(month=month.strftime("%Y-%m-%d"), state_id=STATE_ID)
+                _run_custom_sql_script(sql_to_execute)
+        else:
+            icds_state_aggregation_task(state_id=STATE_ID, date=month,
+                                        func_name='_agg_migration_table')
+
+    def update_ccs_data(self, month):
+        path = os.path.join(os.path.dirname(__file__), 'sql_scripts', 'update_ccs_data.sql')
+        with open(path, "r", encoding='utf-8') as sql_file:
+            sql_to_execute = sql_file.read()
+            sql_to_execute = sql_to_execute.format(month=month.strftime("%Y-%m-%d"), state_id=STATE_ID)
+            sql_to_execute = sql_to_execute.split(';')
+            for i in range(0, len(sql_to_execute)):
+                _run_custom_sql_script(sql_to_execute[i])
+
+    def update_bihar_api_data(self, month):
+        path = os.path.join(os.path.dirname(__file__), 'sql_scripts', 'update_bihar_api_data.sql')
+        with open(path, "r", encoding='utf-8') as sql_file:
+            sql_to_execute = sql_file.read()
+            sql_to_execute = sql_to_execute.format(month=month.strftime("%Y-%m-%d"), state_id=STATE_ID)
+            sql_to_execute = sql_to_execute.split(';')
+            for i in range(0, len(sql_to_execute)):
+                _run_custom_sql_script(sql_to_execute[i])
+
+    def handle(self, *args, **options):
+        start_date = options['start_date'] if options['start_date'] else date(2020, 3, 1)
+        end_date = date(2020, 5, 1)
+        date_itr = start_date
+        while date_itr <= end_date:
+            print(f"Runnning for month {date_itr}=====================\n")
+            self.build_bp_data(date_itr)
+            self.build_pnc_data(date_itr)
+            self.update_ccs_data(date_itr)
+            self.build_migration_forms(date_itr)
```

Comment:
Why are we rebuilding migration forms table?

Reply:
There is a field in the api_demographics added by challa using migration table

Reply:
Not sure, it's better to rebuild to avoid any discrepancy in the data